### ISSUE TITLE
Enhancement: support more than 1 cursor from multiple of pointing devices

### ISSUE THREAD

**Issue by dumblob** [vurtun/zahnrad#26]
Add support for multiple pointing cursors (in text, two cursors are enough - think of mobile devices). It's getting more and more common, that the user has more pointing devices - such az touchpad, trackpoint, external mouse, eye gaze, gamepad, etc.

**Comment by vurtun** [comment]
Hey,
I don't really understand what you mean. Do you mean having multiple ways of providing a "mouse" cursor pointer? 
If so I don't really know what the benefits would be of adding addition data since I don't have the concept of a mouse inside the input struct anyhow. Instead you can use mouse, touchpad, eye gaze,... with the current implementation just fine as long as you provide a current "cursor" position by using the `input_motion` function.

**Comment by dumblob** [comment]
>Do you mean having multiple ways of providing a "mouse" cursor pointer? 

Fortunately not ;)

Right now I started writing very detailed essay about how I imagine the support should look like and then realized, that I didn't write the initial request enough clearly and that if I wrote it right now, there wouldn't be any need to explain anything. So I deleted all the text and wrote only the following sentence.

I meant support for having multiple cursors on the same screen at the same time (e.g. multi-touch in a smartphone to zoom a photo using two fingers).

**Comment by vurtun** [comment]
Ah ok, thanks that makes a lot more sense, then I totally misunderstood. Yeah you are right it is probably worth thinking about ways to provide some mobile input support. Probably have to look into mobile input handling to find the correct abstraction.

**Comment by vurtun** [comment]
So I looked into having multiple pointers and so far it look quite easy to setup. As far as I can see (http://developer.android.com/training/gestures/multi.html) I only have to provide multiple `zr_mouse` (probably have to rename this to `zr_pointer`) and update the `zr_input_motion` and `zr_input_button` function to also provide a way to describe which pointer the even comes from or am I missing something?

**Comment by dumblob** [comment]
Yes, that should be it. There might be some evil in the details, but I can's see any so far.

**Comment by cbuschardt** [comment]
Has anyone explored pen and touch for Windows/Linux at this point?

At least on Windows pen, mouse, and touch compete for ownership of the main cursor.  There are additional APIs to get pen specific or touch specific state.
